src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -854,11 +862,15 @@ def apply_inputs_mapping(
                 result[map_to_key] = map_value  # Literal value
         # Return all not found mapping relations in one exception to provide better debug experience.
         if notfound_mapping_relations:
-            raise MappingSourceNotFound(
-                f"Couldn't find these mapping relations: {', '.join(notfound_mapping_relations)}. "
-                "Please make sure your input mapping keys and values match your YAML input section and input data. "
-                "If a mapping value has a '${data' prefix, it might be generated from the YAML input section, "
-                "and you may need to manually assign input mapping based on your input data."
+            invalid_relations = ", ".join(notfound_mapping_relations)
+            raise InputMappingError(
+                message_format=(
+                    "The input for flow is incorrect. Couldn't find these mapping relations: {invalid_relations}. "
+                    "Please make sure your input mapping keys and values match your YAML input section and input data. "
+                    "If a mapping reads input from 'data', it might be generated from the YAML input section, "
```

Comment:
I might be confused with this expression.

_If a mapping reads input from 'data'_

Reply:
Yes... The previous version used "${data" but met exception as formatting the error message.

Reply:
Add todo to replace the detail message about default mapping by doc link